src/lib/support/AutoRelease.h

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains a standard RAII class which calls Release on dtor.
```

Comment:
I don't understand the use of this in IterateEntries.  There it's passed a pointer to the stack, right?  Why do we need to Release at all if it's going to be destroyed by coming off the stack?

And on the other side: we are passing something that looks refcounted to the provided std::function.  But it's not actually refcounted: once the function returns, it dies, right?  Again, this needs to be clearly documented.

Reply:
This was moved over from `SessionManager.cpp`, you can see its use there

It's not ref-counted, rather allocated somewhere & released on dtor

This is a standard pattern for something to be released in dtor, e.g. something which is allocated from a pool instead of a stack

Matter SDK would likely be a lot easier to manage if we'd used this pattern throughout: instead of stack-allocating, instead use a pool, using `AutoRelease` to release to the pool when done

Added more to the doc, hopefully the comparison to unique_ptr helps?

Reply:
OK.  In that case this all makes sense (though could use some code comments), and we do still need the API documentation explaining that the iterator can only be used during the callback and will get cleaned up automatically.

Reply:
> though could use some code comments
I expanded the doc for the class, was it still unclear? (& if so, what should I add?)

> documentation explaining that the iterator can only be used during the callback
SG, I think that's there right? Or was there more you wanted me to add?
![image](https://github.com/user-attachments/assets/ea177b2d-917a-49a0-8227-36921ac9d386)